Topics/Tech_Stacks/Vue_Component_Setup_and_Tips

```diff
@@ -0,0 +1,73 @@
+
+
+```c
+# Example code
+<script setup>
+import { ref } from 'vue'
+const count = ref(0)
+</script>
+
+<template>
+  <button @click="count++">Count is: {{ count }}</button>
+</template>
+
+<style scoped>
+button {
+  font-weight: bold;
+}
+</style>
+```
+The top is the Javascript code-behind, the middle is the HTML, and the bottom is the CSS style. Nice and centralized. Thankfully, the order doesn't matter. However, as the script is the most complicated, we'll talk about it last.
+
+**Should I explain what the example code above means? Like, what import {ref} does? Or maybe I should give a different example, one similar to what we worked on in our CSC301 project? I worry it might be too long**
+
+### HTML Template
+Ever seen the HTML of a webpage? That and Vue's HTML template are almost identical. Just remember, the root container must be labelled as ```<template>```, and it can only have one container inside that. Inside *that* sub-container though, anything is possible! On top of things like your standard ```<div>```, you can also use **other** Vue Components that you've imported in the script section! More on that in a bit.
+
+Here, I'm going to *only* show how Vue HTML components have if-else visibility features, and the ability to insert components through built-in for loops. Am I missing anything important to do with HTML in Vue?
+
+### CSS
+The CSS styling in Vue works as it does anywhere else, but with one key difference. See that ```scoped``` attribute in the ```<style>``` tag? When the tag has that attribute, the CSS can only apply to the component it's present in. Removing it however, allows for usage anywhere else in the program. Just...be careful about CSS conflicts and shadowing.
+
+### Script
+This is the meat of Vue. This is where you essentially define the webpage as an Object-Oriented-Programming Class! It can have it's own methods, attributes, as well as import scripts and other components for use! It's a very dense part of learning Vue, but here are the basics.
+
+Here I'm going to explain how to mount the app, basically talking about props, the default{} variables, computed methods, import statements, etc. It's going to be a rather dense section, as I have to talk about class variables and how mounted() works at runtime. Should I go into asynchronous methods/components?
+
+## Things to watch out for! <a name="other"></a>
+Vue's reactivity and unique way of handling things can prove to be a double-edged sword if not handled properly. Here are some general tips:
+- If you want to display a document variable inside a component, and that component has an attribute that computes that same variable, you may cause Vue to enter an infinite update loop, where the computing of the variable will cause the displaying of the variable to cause the program to see the component as updated, so it recalculates the variable and redisplays it, and so on. This can crash the page or prevent it from displaying.
+- Uh what else?
```

Comment:
How do we solve this? It'd be handy to know. 
(I believe the solution is to make a child component have a prop).